File: app/src/main/java/com/codemybrainsout/imageviewer/view/photo/PhotoModule.java
package com.codemybrainsout.imageviewer.view.photo;

import android.content.Context;

import com.codemybrainsout.imageviewer.api.PhotoService;
import com.codemybrainsout.imageviewer.dagger.scope.ActivityScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ahulr on 10-06-2017.
 */

@Module
public class PhotoModule {

    private Context context;

    public PhotoModule(Context context) {
        this.context = context;
    }

    @ActivityScope
    @Provides
    Context context() {
        return context;
    }

    @ActivityScope
    @Provides
    PhotoAdapter photoAdapter(Context context) {
        return new PhotoAdapter(context);
    }

    @ActivityScope
    @Provides
    PhotoPresenter photoPresenter(PhotoService photoService) {
        return new PhotoPresenter(photoService, (PhotoContract.View) context);
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/photo/PhotoModule.kt
package com.codemybrainsout.imageviewer.view.photo

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * Created by ahulr on 10-06-2017.
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class PhotoModule {

    @Binds
    abstract fun bindActivity(activity: PhotoActivity): PhotoContract.View

    @Binds
    abstract fun bindPresenter(impl: PhotoPresenter): PhotoContract.Presenter
}